### ISSUE TITLE
Improve support for enum variants

### ISSUE THREAD

**Issue by arendjr** [Aleph-Alpha/ts-rs#22]
I noticed a few issues with enum variants. For instance:

```rs
#[derive(Serialize, TS)]
#[serde(tag = "kind", content = "data")]
enum ComplexEnum {
    A,
    B { foo: String, bar: f64 },
    W(SimpleEnum),
    F { nested: SimpleEnum },
    T(i32, SimpleEnum),
    V(Vec<Series>),
}
```

generates the following type:

```ts
// issue 1: there is no import for the `SimpleEnum` type

export type ComplexEnum =
  | { kind: "A"; data: null } // issue 2: IMO it would be better to omit the `data` property
  | {
    kind: "B";
    data: {
      foo: string;
      bar: number;
    };
  }
  | { kind: "W"; data: SimpleEnum }
  | {
    kind: "F";
    data: {
      nested: SimpleEnum;
    };
  }
  | { kind: "T"; data: [number, SimpleEnum] }
  | { kind: "V"; data: Array }; // issue 3: there is no type argument passed to `Array`
```

I have fixed this issues in my own fork with the following commit: https://github.com/arendjr/ts-rs/commit/e7a87b1b49bfa0263460a63ab995fc133a4ba362

Unfortunately, it would be a bit unwieldy to create a PR for this as this is built on top of both #15 and #18. This is just to let you know these issues are addressed and I can create a PR for it when the other PRs are merged.

**Comment by arendjr** [comment]
As #15 has now been merged, I have included the fixes for these issues in #25, since they partially affected generics support anyway.

**Comment by arendjr** [comment]
Fixed as part of #25.
